qiskit/circuit/library/data_encoding/pauli_expansion.py

```diff
@@ -0,0 +1,216 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Pauli expansion circuit module."""
+
+from typing import Optional, Callable, List, Union
+from functools import reduce
+import numpy as np
+
+from qiskit.circuit import QuantumCircuit
+from qiskit.circuit import Parameter, ParameterVector
+from qiskit.extensions.standard import HGate
+from qiskit.util import deprecate_arguments
+
+from ..n_local.n_local import NLocal
+
+
+class PauliExpansion(NLocal):
+    r"""The Pauli Expansion circuit.
+
+    The Pauli Expansion circuit is a data encoding circuit that transforms input data
+    :math:`\vec{x} \in \mathbb{R}^n` as
+
+    .. math::
+
+        U_{\Phi(\vec{x})}=\exp\left(i\sum_{S\subseteq [n]}
+        \phi_S(\vec{x})\prod_{i\in S} P_i\right)
+
+    The circuit contains ``reps`` repetitions of this transformation.
+    The variable :math:`P_i \in \{ I, X, Y, Z \}` denotes the Pauli matrices.
+    The index :math:`S` describes connectivities between different qubits or datapoints:
+    :math:`S \in \{\binom{n}{k}\ combinations,\ k = 1,... n \}`. Per default the data-mapping
+    :math:`\phi_S` is
+
+    .. math::
+
+        \phi_S(\vec{x}) = \begin{cases}
+            x_0 \text{ if } k = 1 \\
+            \prod_{j \in S} (\pi - x_j)
+
+    Please refer to :class:`FirstOrderExpansion` for the case :math:`k = 1`, :math:`P_0 = Z`
+    and to :class:`SecondOrderExpansion` for the case :math:`k = 2`, :math:`P_0 = Z` and
+    :math:`P_1 P_0 = ZZ`.
+
+    References:
+        [1]: Havlicek et al. (2018), Supervised learning with quantum enhanced feature spaces.
+            https://arxiv.org/abs/1804.11326.
+    """
+
+    @deprecate_arguments({'depth': 'reps'})
+    def __init__(self,
+                 feature_dimension: Optional[int] = None,
+                 reps: int = 2,
+                 entanglement: Union[str, List[List[int]], Callable[[int], List[int]]] = 'full',
+                 paulis: Optional[List[str]] = None,
+                 data_map_func: Optional[Callable[[np.ndarray], float]] = None,
+                 parameter_prefix: str = 'x',
+                 insert_barriers: bool = False,
+                 depth: Optional[int] = None,  # pylint: disable=unused-argument
```

Comment:
since this arg isn't doing anything I think it's safe to remove (in the subclasses too)